### PR TITLE
Highlightjs to v11

### PR DESCRIPTION
**Pull request by dschach**
* Update highlightjs to v11 (there were some security holes noted on the highlight changes list since v9)
* Add Apex highlighting library for optimal code highlights, especially in examples. (Written by @dschach - https://github.com/highlightjs/highlightjs-apex)

### CODE REVIEW

**Comment by no-stack-dub-sack** [comment]
@dschach apologies, it's been a crazy couple of months for me. will try to take a look at these soon.

**Comment by no-stack-dub-sack** [comment]
I promise I'll get to this soon...

File: assets/index.js [hunk 1]
```diff
@@ -113,7 +121,7 @@ function updateMenuModel(items, state) {
 	if (newKeys.length > 0) {
 		newKeys.forEach(item => state[item.id] = item.isOpen === '' && true);
 		console.log('ApexDox: New menu keys found, adding to session storage:');
-		console.log(newKeys.map(function(g) { return g.id }));
+		console.log(newKeys.map(function (g) { return g.id }));
```
Comment (no-stack-dub-sack) [inline assets/index.js:124]: @dschach Is this change related to highlightjs update?

File: assets/index.js [hunk 2]
```diff
@@ -10,7 +10,6 @@ const
 	SEARCH_STATE_KEY = 'APEXDOX_SEARCH_RESULTS';
 
 const highlightJsSelectors = [
```
Comment (no-stack-dub-sack) [inline assets/index.js:12]: @dschach does this need to be updated?

File: .husky/pre-commit [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
```
Comment (no-stack-dub-sack) [inline .husky/pre-commit:1]: I'm wondering if updating highlightJs on every commit may be a little too much. It will also slow down the development workflow (not that much is happening these days) if we have to wait for a network request to complete before every commit. 

What do you think about making this manual? 

A couple of questions to help us answer:
1. How often does this library get updated?
2. How significant are the updates?
3. Is it crucial that it always remain up to date?
4. If this just blindly fetches the latest version of highlight, could we introduce a breaking change unwittingly when a new major version is released?

Comment (dschach) [reply .husky/pre-commit:1]: I think you make a good point here, and I agree with you. We can update it as we test libraries. The Apex library will be updated in a few weeks, so we can make an update here after that.

File: assets/index.js (hunk 2 above)
Comment (dschach) [reply assets/index.js:12]: This block is updated because the JS library requires calling it in new ways. I renamed the variable because these blocks are being specified as Apex. If this extension ever grows to cover LWC or other metadata, we could cover other languages. Or if an org uses functions, we could cover those languages. Apex is just specific here. If I'm way off, I hear you.

Comment (no-stack-dub-sack) [reply assets/index.js:12]: Sorry, I'm ok with the rename - what I mean here was, I think this is a bug, as this line has actually not been updated (I can see how the github suggestion feature is a little confusing, though, because it actually shows the diff of what we want). Within `initHighlightJs` a variable named `apexJsSelectors` is referenced and operated on, however the initialization of that variable does not exist so far as I can tell. This line needs to be updated to the new name.

**Review by no-stack-dub-sack — CHANGES_REQUESTED** [review]
Hey @dschach Thanks for the work on this. Can you please make the changes noted in prev review comments (delete pre-commit highlight update in favor of manual updates and fix var initialization). Once these are fixed I think this should be good to go, but as you mentioned, let's hold of on merging / releasing this until the next highlight update so we can capture the latest version in the next release.

Thanks!

**Comment by dschach** [comment]
Yeah, I'm swamped right now, but I will make time to update the PR and to see what I can get done.

**Comment by no-stack-dub-sack** [comment]
@dschach Absolutely no rush. As you know from working with me on this repo in the past I have a lot going on as well and am not checking in here as regularly as I'd like to be.

**Comment by dschach** [comment]
Changees made. Ready for review.

**Comment by dschach** [comment]
@no-stack-dub-sack Bump on this PR - I'd like to start looking into tslint->eslint and to making a command line interface, once this is merged from my fork. Thanks!

**Comment by no-stack-dub-sack** [comment]
@dschach The PR looks mostly good at this point, thanks for addressing the review comments. Any idea what's up with the yarn.lock file, though? I only see one dependency change (the addition of husky), so would expect a much smaller diff here. Relatedly, I think you can maybe just back out the husky changes now, right? If I remember correctly, highlight JS was previously being updated using husky via the precommit hook, but since you changed it to a manual update process, I think husky is no longer needed, or am I missing maybe something?

**Comment by no-stack-dub-sack** [comment]
Also, I'm assuming you've tested this and everything looks good on your end? If so, since there's no change to core documentation generation code here, I won't feel the need to test this myself assuming you've done so. Thanks and let me know! If we can come to a conclusion on the husky stuff I can get this merged and released within the week.

**Comment by dschach** [comment]
I've tested it with these versions on a few of my repos by overriding the library you've bundled. It works well, as far as I can see. Check any of my repos and they're done with 11.7.

Husky won't do anything yet, but we can install it now and it won't run anything if it doesn't have anything in its hooks. 

So yeah, it's tested :)

**Comment by no-stack-dub-sack** [comment]
@dschach Thanks - I think my preference at this point would be to remove husky (and the changes to the lock file) until we need it for something. When we do, it can be installed and go in as a part of that pull request.

**Comment by dschach** [comment]
Removed. PR is only to update highlight and the new index.js file

**Comment by no-stack-dub-sack** [comment]
@dschach reverted lockfile changes and removed the `prepare` script from `package.json`

**Comment by no-stack-dub-sack** [comment]
Will get this released soon. Would like to give it a few days to see if #68 materializes and if they can go out as part of the same release if so.
